plugins/modules/lvm_facts.py

```diff
@@ -329,6 +330,16 @@ def load_vgs(module, name, LVM):
             rc, out, err = module.run_command(cmd)
             if rc != 0:
                 msg += "Command '%s' failed." % cmd
+                # make sure that varied off volume groups
+                # are returned.
+                # 0516-010: Volume group must be varied on; use varyonvg command.
+                pattern = r"0516-010"
+                found = re.search(pattern, err)
+                if found:
+                    data = {
+                        'vg_state': "deactivated"
+                    }
+                    LVM['VGs'][vg] = data
```

Comment:
Where is LVM defined? It is a dict, but can not find it definitions. is it a build-in?

Reply:
LVM is defined in main function. LVM defaults to an empty dict {}
then, when passed to appropriate handlers, it will add dictionaries for that component.
i.e when fetching for VGs , then LVM will have LVM[VGs] = {} defined. and so on and so forth